core/domain/user_query_domain.py

```diff
@@ -19,26 +19,46 @@
 from __future__ import annotations
 
 import collections
+import datetime
 
 from core import feconf
 from core import utils
-from core.constants import constants
 
-attribute_names = [ # pylint: disable=invalid-name
-    predicate['backend_attr'] for predicate in (
-        constants.EMAIL_DASHBOARD_PREDICATE_DEFINITION)]
+from typing import List, Optional
 
-UserQueryParams = collections.namedtuple( # pylint: disable=invalid-name
-    'UserQueryParams', attribute_names, defaults=(None,) * len(attribute_names))
+
+# We can't inject dynamic data (attribute_names) in namedtuple (UserQueryParams)
+# without raising MyPy error. Therefore, we need to use predefined attributes
+# explicitly and then verify them in the test file and raise errors if needed.
```

Comment:
I don't think this comment is needed anymore.

Reply:
Sure !!